torch/csrc/jit/passes/create_autodiff_subgraphs.cpp

```diff
@@ -110,21 +110,26 @@ class SubgraphSlicer {
     return result;
   }
 
-  bool shouldConsiderForMerge(Node* node) {
+  bool shouldConsiderForMerge(Node* node, const AliasDb& aliasDb) {
     // if we're already in the process of merging
     if (node->kind() == prim::DifferentiableGraph) {
       return true;
     }
     if (node->kind() == prim::Constant) {
       return false;
     }
+    // when a node which has writers is moved into a subgraph it may lose
+    // context and CSE could merge it with another node that has writers
```

Comment:
It would be good to reference the discussion from the pull request, add a FIXME, and say that we're going to fix the problem more generally for CSE.